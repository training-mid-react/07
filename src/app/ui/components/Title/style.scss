@use '../../styles/variables' as var;

.title {
    color: var.$color-white
}