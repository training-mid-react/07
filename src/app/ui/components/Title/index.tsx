import React from 'react';
import './style.scss';

interface TitleProps {
  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  children: React.ReactNode;
  className?: string;
}

const Title: React.FC<TitleProps> = ({ as: Tag, children, className }) => {
  return <Tag data-testid="title-component" className={className || 'title'}>{children}</Tag>;
};

export default Title;
