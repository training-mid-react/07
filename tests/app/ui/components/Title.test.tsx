import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Title from '../../../../src/app/ui/components/Title';

describe('Componente Title', () => {
  it('renderiza correctamente como h1', () => {
    const { asFragment } = render(<Title as="h1">Título H1</Title>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renderiza correctamente como h2', () => {
    const { asFragment } = render(<Title as="h2">Título H2</Title>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renderiza correctamente con className personalizado', () => {
    const { asFragment } = render(
      <Title as="h3" className="custom-title">
        Título H3
      </Title>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renderiza con el contenido correcto', () => {
    const { getByTestId } = render(<Title as="h4">Contenido del título</Title>);
    const titleElement = getByTestId('title-component');
    expect(titleElement).toHaveTextContent('Contenido del título');
  });

  it('aplica la clase personalizada correctamente', () => {
    const { getByTestId } = render(<Title as="h5" className="custom-class">Título H5</Title>);
    const titleElement = getByTestId('title-component');
    expect(titleElement).toHaveClass('custom-class');
  });

  it('aplica la clase por defecto cuando no se pasa className', () => {
    const { getByTestId } = render(<Title as="h6">Título H6</Title>);
    const titleElement = getByTestId('title-component');
    expect(titleElement).toHaveClass('title');
  });
});
