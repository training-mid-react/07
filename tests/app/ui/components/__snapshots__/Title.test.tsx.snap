// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Title > renderiza correctamente como h1 1`] = `
<DocumentFragment>
  <h1
    class="title"
    data-testid="title-component"
  >
    Título H1
  </h1>
</DocumentFragment>
`;

exports[`Componente Title > renderiza correctamente como h2 1`] = `
<DocumentFragment>
  <h2
    class="title"
    data-testid="title-component"
  >
    Título H2
  </h2>
</DocumentFragment>
`;

exports[`Componente Title > renderiza correctamente con className personalizado 1`] = `
<DocumentFragment>
  <h3
    class="custom-title"
    data-testid="title-component"
  >
    Título H3
  </h3>
</DocumentFragment>
`;
